sklearn/manifold/tests/test_t_sne.py

```diff
@@ -296,10 +305,12 @@ def test_preserve_trustworthiness_approximately(method, init):
     assert t > 0.85
```

Comment:
`TSNE` is not in the list of #11000 and `check_transformer_preserve_dtypes` is not called on it because `TSNE` does not have a `transform` method (only `fit_transform`) and therefore is not a "real" transformer.

So I think that we should manually test in this file, at least in one of the tests of this file that the output of `fit_transform` preserves the dtype:

```suggestion
    assert t > 0.85
    assert X_embedded.dtype == dtype
```

I have checked locally and this should always hold.

Note: I cannot make the suggestion on a new line right after the definition of the `X_embedded` variable.

Reply:
Actually this will not work. Apparently our code always works with float32 precision. I think this should be documented in the docstring as this is behavior is quite specific to this estimator.

Reply:
It does preserves dtypes when we provide `init` to TSNE as a float64 numpy array, but in other cases it does not.

Everything depends on the initialisation of `X_embedded`:
https://github.com/scikit-learn/scikit-learn/blob/ae98150e411bb4dc00822d935d8341762b4323fc/sklearn/manifold/_t_sne.py#L967-L993

Reply:
Indeed, we should be documenting this behaviour in the docstring.

Reply:
Done in the `Notes` section of the docstring of TSNE.